src/pages/question/questionBuilder/QuestionBuilder.jsx

```diff
@@ -71,30 +76,48 @@ export default function QuestionBuilder(props) {
     toggleModal(false);
   }
 
+  function validateAndParseFile(fileContents) {
+    let fileContentObj;
+    try {
+      fileContentObj = JSON.parse(fileContents);
+    } catch (err) {
+      displayAlert('error',
+        'The provided file is not valid JSON. Please fix and try again.');
+      setStep('options');
+      return;
+    }
+
+    if (!('query_graph' in fileContentObj)) {
+      displayAlert('error',
+        'The provided file does not have a query_graph object. Please ensure that the file format adheres to the Reasoner Standard API Query Schema.');
+      setStep('options');
+      return;
+    }
+
+    if (!_.isArray(fileContentObj.query_graph.nodes) ||
+       !_.isArray(fileContentObj.query_graph.edges)) {
+      displayAlert('error',
+        'The provided file does not have an array of nodes or edges. Please ensure that the file format adheres to the Reasoner Standard API Query Schema.');
+      setStep('options');
+      return;
+    }
+
+    questionStore.loadListRepresentation(
+      fileContentObj.query_graph,
+    );
+    setStep('build');
+    setLoading(false);
+  }
+
   function onDropFile(acceptedFiles, rejectedFiles) { // eslint-disable-line no-unused-vars
     acceptedFiles.forEach((file) => {
       const fr = new window.FileReader();
       fr.onloadstart = () => setLoading(true);
       // fr.onloadend = () => this.setState({ graphState: graphStates.fetching });
-      fr.onload = (e) => {
-        const fileContents = e.target.result;
-        try {
-          const fileContentObj = JSON.parse(fileContents);
-          questionStore.loadListRepresentation(fileContentObj);
-          setStep('build');
-          setLoading(false);
-        } catch (err) {
-          console.error(err);
-          // window.alert('Failed to read this Question template. Are you sure this is valid?');
-          // questionStore.setGraphState(graphStates.error);
-          setLoading(false);
-        }
-      };
+      fr.onload = (e) => validateAndParseFile(e.target.result);
       fr.onerror = () => {
-        window.alert('Sorry but there was a problem uploading the file. The file may be invalid JSON.');
-        questionStore.resetQuestion();
-        setLoading(false);
-        // questionStore.setGraphState(graphStates.error);
+        displayAlert('error', 'Sorry but there was a problem uploading the file. Please try again.');
+        setStep('options');
```

Comment:
From a code perspective, I would think we want a setLoading(false); here, but it doesn't seem to matter. So either the fr.onloadstart function is not run when I think it is, or the alert is rerendering the page. Don't know if we need to do anything, but this seems weird.